packages/oauth-client/src/EarthdataLoginClient.ts

```diff
@@ -0,0 +1,180 @@
+import got, { HTTPError, Response } from 'got';
+
+import { OAuthClient } from './OAuthClient';
+import { EarthdataLoginError } from './EarthdataLoginError';
+
+type VerifyTokenResponse = Response<{uid: string}>;
+
+type EarthdataLoginErrorResponse = Response<{error: string}>;
+
+const isHttpForbiddenError = (error: unknown) =>
+  error instanceof HTTPError && error.response.statusCode === 403;
+
+const isHttpBadRequestError = (error: unknown) =>
+  error instanceof HTTPError && error.response.statusCode === 400;
+
+const httpErrorToEarthdataLoginError = (httpError: HTTPError) => {
+  const response = <EarthdataLoginErrorResponse>httpError.response;
+
+  switch (response.body.error) {
+    case 'invalid_token':
+      return new EarthdataLoginError('InvalidToken', 'Invalid token');
+    case 'token_expired':
+      return new EarthdataLoginError('TokenExpired', 'The token has expired');
+    default:
+      return new EarthdataLoginError(
+        'UnexpectedResponse',
+        `Unexpected response: ${httpError.response.body}`
+      );
+  }
+};
```

Comment:
This is the existing error handling I mentioned in previous comments. It's moved to a new file but otherwise unchanged.